### ISSUE TITLE
CSharpCodeGenerator---FromFileBatch Implementation request

### ISSUE THREAD

**Issue by MockBird0001** [dotnet/runtime#70817]
I am preparing to generate a dll file from user code, I use the "CodeDomProvider.CompileAssemblyFromSource()" method, but the report shows that it cannot be used on the NetCore platform, I checked the donet source code and found that this is a feature that is not implemented (https://github.com/dotnet/runtime/blob/3c33b4293107c607ae8669545114d6f9891432cb/src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.PlatformNotSupported.cs), or Where else can I implement the user code generation dll.

Thanks

**Comment by stephentoub** [comment]
The recommended way to do this is using APIs from the C# compiler (Roslyn).  .NET Framework ships with csc.exe as part of the installation, and CodeDOM';s CSharpCodeGenerator just shell'd out to that csc.exe; .NET Core doesn't ship with csc.exe, so CodeDOM can't do that.  On top of which, it's not a particularly efficient approach.  Better to just use the APIs the C# compiler exposes for doing this task.

e.g.
https://stackoverflow.com/questions/32769630/how-to-compile-a-c-sharp-file-with-roslyn-programmatically

**Comment by Tianfei-Wang** [comment]
Hi guys, I looked at the above and tested it, but I have new questions about source generators:

My source generator project needs to reference a Net6.0 project (CommonB), and reference the CommonB namespace in the SourceText of the source generator and override the abstract methods of the abstract classes in CommonB.
But I don't know how to properly reference CommonB in the source generator project, I tried directly referencing its dll, or packaging CommonB into Nuget and then referencing, or even referencing the project directly,
But this referenced project always seems to be not found when the user code in the source generator works.

My questions: 
1. How should I reference other projects in the source generator project like to CodeDom.Compiler.CompilerParameters.ReferencedAssemblies.Add(CommonB)?
2. The source generator is based on Stan2.0, and the reference project is Net6.0. Are they incompatible?
